service/src/test/java/uk/nhs/adaptors/gp2gp/gpc/GpcTaskAggregateServiceTest.java

```diff
@@ -0,0 +1,66 @@
+package uk.nhs.adaptors.gp2gp.gpc;
+
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+import java.util.Collections;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import uk.nhs.adaptors.gp2gp.ehr.SendEhrExtractCore;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus;
+
+@ExtendWith(MockitoExtension.class)
+public class GpcTaskAggregateServiceTest {
+
+    @Mock
+    private SendEhrExtractCore sendEhrExtractCore;
+
+    private GpcTaskAggregateService gpcTaskAggregateService;
+
+    @BeforeEach
+    public void setUp() {
+        gpcTaskAggregateService = new GpcTaskAggregateService(sendEhrExtractCore);
+    }
+
+    @Test
+    public void When_AllPreparingDataStepsAreFinished_Expect_SendEhrExtractTaskCreated() {
+        EhrExtractStatus ehrExtractStatus = getEhrExtractStatus("object_name");
+
+        gpcTaskAggregateService.sendData(ehrExtractStatus);
+
+        verify(sendEhrExtractCore).send(ehrExtractStatus);
+    }
+
+    @Test
+    public void When_AllPreparingDataStepsAreNotFinished_Expect_SendEhrExtractTaskNotCreated() {
+        EhrExtractStatus ehrExtractStatus = getEhrExtractStatus(null);
+
+        gpcTaskAggregateService.sendData(ehrExtractStatus);
+
+        verify(sendEhrExtractCore, never()).send(ehrExtractStatus);
+    }
+
+    private EhrExtractStatus getEhrExtractStatus(String objectName) {
```

Comment:
I would name it "prepare", or "build" instead of "get". It suggests that we are invoking a getter method.